tso-server/server/server.go

```diff
@@ -41,46 +49,110 @@ type atomicObject struct {
 
 // TimestampOracle generates a unique timestamp.
 type TimestampOracle struct {
-	ts     atomic.Value
-	ticker *time.Ticker
+	cfg *Config
+
+	ts atomic.Value
 
 	listener net.Listener
 
 	wg       sync.WaitGroup
 	connLock sync.Mutex
 	conns    map[net.Conn]struct{}
+
+	zkElector *zkhelper.ZElector
+	zkConn    zkhelper.Conn
+
+	isLeader int64
+
+	lastSavedTime time.Time
 }
 
-func (tso *TimestampOracle) updateTicker() {
-	// TODO: save latest TS to persistent storage(zookeeper/etcd...)
+func (tso *TimestampOracle) loadTimestamp() error {
+	last, err := loadTimestamp(tso.zkConn, tso.cfg.RootPath)
+	if err == zk.ErrNoNode {
```

Comment:
no need, loadTimestamp handles ErrNoNode first and returns zk.ErrNoNode directly.